### ISSUE TITLE
CutOut augmentation

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#39]

**Comment by bhack** [comment]
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/cutout_ops.py

https://github.com/tensorflow/models/blob/master/official/vision/image_classification/augment.py#L267

**Comment by innat** [comment]
@LukeWood There is another augmentation similar to this one called [CoarseDropout](https://albumentations.ai/docs/api_reference/augmentations/dropout/coarse_dropout/#albumentations.augmentations.dropout.coarse_dropout.CoarseDropout). I think it's somewhat better to have than simple CutOut. 

[TF Code. ](https://www.kaggle.com/cdeotte/tfrecord-experiments-upsample-and-coarse-dropout)

![output_13_0](https://user-images.githubusercontent.com/17668390/150628277-fde78218-99d5-41a1-a060-7c9d98e04c1c.png)

Here is another variant on the cutout, called [progressive sprinkles](https://twitter.com/jeremyphoward/status/1150927513666953216)

**Comment by LukeWood** [comment]
Thanks @chjort
